clients/filesystem-fuse/src/fuse_api_handle.rs

```diff
@@ -30,10 +30,474 @@ use fuse3::{Errno, FileType, Inode, SetAttr, Timestamp};
+            .field("mtime", &TimestampDebug(attr.mtime))
+            .field("ctime", &TimestampDebug(attr.ctime));
+
+        // Conditionally add the "crtime" field only for macOS
+        #[cfg(target_os = "macos")]
+        {
+            struc.field("crtime", &TimestampDebug(attr.crtime));
+        }
+
+        struc
+            .field("kind", &attr.kind)
+            .field("perm", &attr.perm)
+            .field("nlink", &attr.nlink)
+            .field("uid", &attr.uid)
+            .field("gid", &attr.gid)
+            .field("rdev", &attr.rdev)
+            .finish()
+    }
+}
+
+pub(crate) struct FuseApiHandleDebug<T: RawFileSystem> {
+    inner: FuseApiHandle<T>,
+}
+
+impl<T: RawFileSystem> FuseApiHandleDebug<T> {
+    pub fn new(fs: T, context: FileSystemContext) -> Self {
+        Self {
+            inner: FuseApiHandle::new(fs, context),
+        }
+    }
+}
+
+impl<T: RawFileSystem> Filesystem for FuseApiHandleDebug<T> {
+    async fn init(&self, req: Request) -> fuse3::Result<ReplyInit> {
+        debug!("init [id={}]: req: {:?}", req.unique, req);
+        let res = self.inner.init(req).await;
+        match res {
+            Ok(reply) => {
+                debug!("init [id={}]: reply: {:?}", req.unique, reply);
+                Ok(reply)
+            }
+            Err(e) => {
+                debug!("init [id={}]: error: {:?}", req.unique, e);
+                Err(e)
+            }
+        }
+    }
+
+    async fn destroy(&self, req: Request) {
+        debug!("destroy [id={}]: req: {:?}", req.unique, req);
+        self.inner.destroy(req).await;
+        debug!("destroy [id={}]: completed", req.unique);
+    }
+
+    async fn lookup(&self, req: Request, parent: Inode, name: &OsStr) -> fuse3::Result<ReplyEntry> {
+        debug!(
+            "lookup [id={}]: req: {:?}, parent: {:?}, name: {:?}",
+            req.unique, req, parent, name
+        );
+        let result = self.inner.lookup(req, parent, name).await;
+        match result {
+            Ok(reply) => {
+                debug!("lookup [id={}]: reply: {:?}", req.unique, reply);
+                Ok(reply)
+            }
+            Err(e) => {
+                debug!("lookup [id={}]: error: {:?}", req.unique, e);
+                Err(e)
+            }
+        }
+    }
+
+    async fn getattr(
+        &self,
+        req: Request,
+        inode: Inode,
+        fh: Option<u64>,
+        flags: u32,
+    ) -> fuse3::Result<ReplyAttr> {
+        debug!(
+            "getattr [id={}]: req: {:?}, inode: {:?}, fh: {:?}, flags: {:?}",
+            req.unique, req, inode, fh, flags
+        );
+        let result = self.inner.getattr(req, inode, fh, flags).await;
+        match result {
+            Ok(reply) => {
+                debug!(
+                    "getattr [id={}]: reply: {:?}",
+                    req.unique,
+                    FileAttrDebug {
+                        file_attr: &reply.attr
+                    }
```

Comment:
Constructing an object is not as clear as using  a function like `to_desc_string()`